### PR TITLE
Feature/#153 document dev setup

### PR DESCRIPTION
**Pull request by TheGreatRefrigerator**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,29 +30,37 @@ See also:
...
 API key: https://openrouteservice.org/sign-up/
 
 ### Installation
 
-Either from QGIS plugin repository or manually:
+In the QGIS menu bar click `Plugins` ► `Manage and Install Plugins...`.
+
+Then search for `openrouteservice` and install `ORS Tools`.
+
+Or install the plugin manually:
```
Comment (koebi) [inline README.md:63]: Replace the "Or" with "Alternatively, you can " or something else

File: README.md [hunk 2]
```diff
@@ -61,10 +69,11 @@ Either from QGIS plugin repository or manually:
...
+This project is published under the GPLv3 license, see [LICENSE.md](https://github.com/GIScience/orstools-qgis-plugin/blob/main/LICENSE.md) for details.
 
-By using this plugin, you also agree to the terms and conditions of openrouteservice, as outlined [here](https://openrouteservice.org/terms-of-service/).
+By using this plugin, you also agree to the [terms and conditions](https://openrouteservice.org/terms-of-service/) of
+openrouteservice.
 
 ## Acknowledgements
 
-This project was first started by [Nils Nolde](https://github.com/nilsnolde), [GIS • OPS UG](https://gis-ops.com).
+This project was first started by [Nils Nolde](https://github.com/nilsnolde).
```
Comment (koebi) [inline README.md:272]: Is there a reason for this deletion?

File: README.md [hunk 3]
```diff
@@ -48,11 +48,11 @@ in the GUI.
 
 ## Getting Started
 
-### Prerequisites
+### Requirements
 
 QGIS version: **v3.4** or above
 
-API key: https://openrouteservice.org/sign-up/
+ORS API key: https://openrouteservice.org/sign-up/
```
Comment (koebi) [inline README.md:55]: This should be a clickable link

File: README.md [hunk 4]
```diff
@@ -67,6 +67,199 @@ Or install the plugin manually:
     - Windows: `C:\Users\USER\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins`
     - Mac OS: `Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins`
 
+## Development Setup
+
+### Requirements:
+- QGIS-LTR (3.16)
```
Comment (koebi) [inline README.md:73]: Does this also work with older QGIS versions? Given the tools minimal requirement of 3.4 - it should?

File: README.md [hunk 5]
```diff
@@ -67,6 +67,199 @@ Or install the plugin manually:
...
+  
+  plugins:
```
Comment (koebi) [inline README.md:75]: _Recommended_ Plugins

File: README.md [hunk 6]
```diff
@@ -67,6 +67,199 @@ Or install the plugin manually:
...
+
+### Interface development
+
+For designing the Dialog the Qt designer shipping with qgis is used. It has relevant classes such as
+[QgsMapLayerComboBox](https://qgis.org/pyqgis/3.2/gui/Map/QgsMapLayerComboBox.html) already imported properly.
+
+#### Mac
+- use `/Applications/QGIS-LTR.app/Contents/MacOS/bin/designer` instead of
+  `/Applications/QGIS-LTR.app/Contents/MacOS/Designer.app` (which is not working, and trying to get other Qt Designer
+  or Qt Creator installations to use the correct QGIS classes is a nightmare that cost me 3 days of my life)
```
Comment (koebi) [inline README.md:229]: While this is a fun story to tell - it shouldn't be mentioned here, at least not like this ;)

File: README.md [hunk 7]
```diff
@@ -67,6 +67,199 @@ Or install the plugin manually:
...
+    ```shell
+    cd /Applications
+    ln -s QGIS-LTR.app/Contents/MacOS/bin/designer "Qt Designer.app"
+    ```
+
+#### Windows
+- should create you a shortcut to the Qt Designer with the installation
+
+#### Linux
+- to be determined
```
Comment (koebi) [inline README.md:240]: You can use `designer` to open anything :)

Comment (koebi) [reply README.md:240]: Instructions are in the wiki, I think?

**Review by koebi — CHANGES_REQUESTED** [review]
Few comments, rest lgtm.
Didn't verify anything pycharm related though ;)

File: README.md (hunk 4 above)
Comment (TheGreatRefrigerator) [reply README.md:73]: This is the QGIS installation used during development, which should be at least the LTR version.
Compatibility should go down to the minimal requirement, but development should be done on LTR and in the best case also have upcoming changes in mind (3.22 available already). wdyt?

File: README.md (hunk 2 above)
Comment (TheGreatRefrigerator) [reply README.md:272]: yes, we should not (are not allowed to) promote external companies. Name of the author should be sufficient.

File: README.md (hunk 6 above)
Comment (TheGreatRefrigerator) [reply README.md]: rephrased

File: README.md (hunk 5 above)
Comment (TheGreatRefrigerator) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (TheGreatRefrigerator) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (TheGreatRefrigerator) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (koebi) [reply README.md:272]: :+1:

File: README.md (hunk 4 above)
Comment (koebi) [reply README.md:73]: ah yes, we're talking about the dev setup here. True, I agree.

**Review by koebi — APPROVED** [review]

**Pull request merged**